### ISSUE TITLE
Archers and Shivas can fire through blocks

### ISSUE THREAD

**Issue by Enduar** [Blaylock1988/VanillaReplacement-GV-Weapons-Pack#34]
Missiles can apparently fire through blocks covering their launch ports- this makes them pretty OP since you can completely armor/conceal them

**Comment by Blaylock1988** [comment]
#38

**Comment by Blaylock1988** [comment]
Somehow adding the option for ToggleGuidance, and then toggling that on the block suddenly made the missiles stop firing through blocks. Need to look into this further.

**Comment by Blaylock1988** [comment]
I've learned this is actually an intended feature. This was done because homing missiles would often do a 180 and fly back into the ship that launched them (because they have no collision avoidance). Supposedly the weapon model needs to have a "Scope" empty added in order for it to do a proper detection and make sure it is open to the outside (not covered up) but this is not on the current Archer missile pods model. The workaround in the meantime has been to add a 2nd stage for the homing, and the 1st stage is now a non-guided ammo

**Comment by Blaylock1988** [comment]
https://github.com/sstixrud/WeaponCore/issues/330

**Comment by Enduar** [comment]
This seems odd considering you have to account for weapon/turret facing with every other system. Just don't fire missiles at a target if they're on the back side of the ship relative to the launchers- seems like an easy enough thing for the user to take into account. The alternative being that you're able to completely bury missiles under armor seems like a far worse issue TBH and not worth the trade.

**Comment by Blaylock1988** [comment]
I agree but it's not my call, and looks like they aren't changing it so we
gotta deal with workarounds.
